tests/test_core.py

```diff
@@ -548,74 +579,80 @@ def assert_bad_tcr_ecs_diagnosis_values_caught(
+    assert_bad_tcr_ecs_tcre_diagnosis_values_caught(
         test_RF_data,
-        magicc_base._check_tcr_ecs_total_RF,
-        r"The TCR/ECS total radiative forcing looks wrong.*",
-        valid_tcr_ecs_diagnosis_results["tcr_time"],
-        valid_tcr_ecs_diagnosis_results["ecs_time"],
+        magicc_base._check_tcr_ecs_tcre_total_RF,
+        r"The TCR/ECS/TCRE total radiative forcing looks wrong.*",
+        valid_tcr_ecs_tcre_diagnosis_results["tcr_time"],
+        valid_tcr_ecs_tcre_diagnosis_results["ecs_time"],
     )
 
 
-def test_check_tcr_ecs_temp(valid_tcr_ecs_diagnosis_results, magicc_base):
-    test_results_df = valid_tcr_ecs_diagnosis_results["mock_results"]
+def test_check_tcr_ecs_tcre_temp(valid_tcr_ecs_tcre_diagnosis_results, magicc_base):
+    test_results_df = valid_tcr_ecs_tcre_diagnosis_results["mock_results"]
     test_temp_data = test_results_df.filter(variable="Surface Temperature")
 
-    magicc_base._check_tcr_ecs_temp(test_temp_data)
+    magicc_base._check_tcr_ecs_tcre_temp(test_temp_data)
 
-    assert_bad_tcr_ecs_diagnosis_values_caught(
+    assert_bad_tcr_ecs_tcre_diagnosis_values_caught(
         test_temp_data,
-        magicc_base._check_tcr_ecs_temp,
-        r"The TCR/ECS surface temperature looks wrong, it decreases",
+        magicc_base._check_tcr_ecs_tcre_temp,
+        r"The TCR/ECS/TCRE surface temperature looks wrong, it decreases",
         test_target="temperature",
     )
 
 
 # integration test (i.e. actually runs magicc) hence slow
 @pytest.mark.slow
-def test_integration_diagnose_tcr_ecs(package):
-    actual_result = package.diagnose_tcr_ecs()
+def test_integration_diagnose_tcr_ecs_tcre(package):
+    actual_result = package.diagnose_tcr_ecs_tcre()
     assert isinstance(actual_result, dict)
     assert "tcr" in actual_result
     assert "ecs" in actual_result
+    assert "tcre" in actual_result
     assert actual_result["tcr"] < actual_result["ecs"]
     if isinstance(package, MAGICC6):
-        assert (
-            actual_result["tcr"] == 1.9733976000000002
-        )  # MAGICC6 shipped with pymagicc should be stable
-        assert (
-            actual_result["ecs"] == 2.9968448
-        )  # MAGICC6 shipped with pymagicc should be stable
+        # MAGICC6 shipped with pymagicc should be stable
+        np.testing.assert_allclose(actual_result["tcr"], 1.9733976)
+        np.testing.assert_allclose(actual_result["ecs"], 2.9968448)
+        np.testing.assert_allclose(actual_result["tcre"], 0.00229016699)
+
     if isinstance(package, MAGICC7):
         # see how stable this is, can delete the test later if it's overly restrictive
-        assert actual_result["tcr"] == 2.0875821584565704
-        assert actual_result["ecs"] == 3.155772857010324
+        np.testing.assert_allclose(actual_result["tcr"], 1.982697)
+        np.testing.assert_allclose(actual_result["ecs"], 2.9948422)
+        np.testing.assert_allclose(actual_result["tcre"], 0.0023189736)
```

Comment:
That's a 1% difference for TCRE between M6 and M7. Is that just differences in defaults

Reply:
Yep and new ocean parameterisation etc.